src/errors.rs

```diff
@@ -504,6 +505,45 @@ impl Display for S3ErrorCode {
         <Self as fmt::Debug>::fmt(self, f)
     }
 }
+
+/// S3 error code parse error
+#[allow(clippy::exhaustive_enums)]
+pub enum ParseS3ErrorCodeError {
+    /// String error code parsing error
+    FromStr(String),
+}
```

Comment:
I suggest that the new error type should follow the existing errors by default. Users may not expect an extra allocation if they don't care whether the error carries the input string.

https://github.com/datenlord/s3-server/blob/5451f0a273bc0503212a7583714cc7d05d4e357a/src/headers/authorization_v4.rs#L117-L124

Reply:
Unfortunately clippy runs into an ICE in CI. I don't know why it happens.

Reply:
changed it to store &str instead of String to avoid the allocation. I do think showing the actual error code message to the user will help with debugging.

Reply:
The clippy crash is caused to upstream bug. The code base enables all clippy rules by default, some of the non-default clippy rules are not well tested. We should report the crash upstream.